scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1342,10 +1344,8 @@ private void sendMessage(final RawData message, final DTLSSession session) {
 					session.getSequenceNumber(),
 					new ApplicationMessage(message.getBytes(), message.getInetSocketAddress()),
 					session);
-			if (message.getMessageCallback() != null) {
-				message.getMessageCallback().onContextEstablished(ctx);
-			}
```

Comment:
Even if the API look better, I have some concerns about correlation context is now sent to upper layer only after record was sent. I think this could involve some race condition issues.  WDYT ?